mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -664,13 +665,25 @@ LogicalResult ThreadwiseReadIntoRewritePattern::matchAndRewrite(
   auto [buffer, transforms] = untransform(b, sourceView, op.getExtraViews());
 
   int64_t numValues = dest.getType().getNumElements();
-  ArrayRef<int64_t> bufferShape =
-      buffer.getType().cast<ShapedType>().getShape();
+  MemRefType srcBufferType = buffer.getType().cast<MemRefType>();
+  ArrayRef<int64_t> bufferShape = srcBufferType.getShape();
+
+  // Unless specified it is assumed to be global
+  gpu::AddressSpace srcAddrSpace = gpu::AddressSpace::Global;
+  if (srcBufferType.getMemorySpace()) {
+    srcAddrSpace =
+        srcBufferType.getMemorySpace().cast<gpu::AddressSpaceAttr>().getValue();
+  }
+  DenseMap<gpu::AddressSpace, size_t> iterDim{
```

Comment:
In silly minor notes, I think there's a `SmallDenseMap` we could use here? And I wonder if we can `constexpr` this